Pick fixes on top of 8.5.0.beta1

tsuoanttila opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/11020)
<!-- Reviewable:end -->

tsuoanttila committed:
Fix ComboboxScrollableWindowTest to use assertion (#11014)

tsuoanttila committed:
Add alternative screenshot for Flash test (#11013)

tsuoanttila committed:
Fix Firefox test issues (#11017)

tsuoanttila committed:
Add pluginRepository to test/pom.xml (#11015)

tsuoanttila committed:
Fix expanding an added TreeGrid node (#11018)

Fixes #9997

elmot commented:
Reviewed 5 of 5 files at r1, 1 of 1 files at r2, 56 of 56 files at r3, 1 of 1 files at r4, 3 of 3 files at r5.
Review status: all files reviewed, 1 unresolved discussion

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/framework/11020)*
<!-- Sent from Reviewable.io -->

tsuoanttila committed:
Fix wrapping div getting focused in CheckBoxGroup and RadioButtonGroup (#11021)

Fixes #9731

tsuoanttila committed:
Fix Firefox coordinates, add missing screenshots (#11022)

Commit:
Fix formatting (#11023)

elmot reviewed (approved):
> Reviewed 3 of 3 files at r6, 42 of 42 files at r7, 7 of 7 files at r8.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11020)* status: all files reviewed, 1 unresolved discussion



<!-- Sent from Reviewable.io -->

tsuoanttila committed:
Enable selection checkboxes with unbuffered editor (#11032)

Fixes #11028

mehdi-vaadin committed:
Improve ComboBox keyboard behavior when an item is selected (#11031)

Ignoring filtering options when right arrow, left arrow, home or end key is pressed.

Fixes #11016

tsuoanttila committed:
Fix initially selected CheckBox rendering in Safari (#11024)

elmot reviewed (approved):
> Reviewed 1 of 1 files at r9, 3 of 3 files at r10, 1 of 1 files at r11.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11020)* status: all files reviewed, 1 unresolved discussion



<!-- Sent from Reviewable.io -->

elmot reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/11020)* status: all files reviewed, 1 unresolved discussion



<!-- Sent from Reviewable.io -->
